### PR TITLE
[ITSASU-1241] Redirect to task list when no businesses can be added.

### PR DESCRIPTION
**Pull request by aug24**

### CODE REVIEW

File: app/controllers/agent/WhatIncomeSourceToSignUpController.scala [hunk 1]
```diff
@@ -46,8 +46,9 @@ class WhatIncomeSourceToSignUpController @Inject()(val whatIncomeSourceToSignUp:
     implicit user =>
       if (isEnabled(SaveAndRetrieve)) {
         withAgentReference { reference =>
-          withIncomeSourceStatuses(reference) { incomeSourcesStatus =>
-            Ok(view(businessIncomeSourceForm(incomeSourcesStatus), incomeSourcesStatus))
+          withIncomeSourceStatuses(reference) {
```
Comment (aug24) [inline app/controllers/agent/WhatIncomeSourceToSignUpController.scala:49]: This is the main change.

File: app/controllers/agent/WhatIncomeSourceToSignUpController.scala [hunk 2]
```diff
@@ -102,7 +103,11 @@ class WhatIncomeSourceToSignUpController @Inject()(val whatIncomeSourceToSignUp:
   }
 
   private def overseasPropertyAvailable(reference: String)(implicit hc: HeaderCarrier): Future[Boolean] = {
-    subscriptionDetailsService.fetchOverseasProperty(reference) map (_.isEmpty && isEnabled(ForeignPropertyFeature))
+    if (isEnabled(ForeignPropertyFeature)) {
```
Comment (aug24) [inline app/controllers/agent/WhatIncomeSourceToSignUpController.scala:106]: This is slightly more efficient but also much easier to understand.

File: it/controllers/agent/WhatIncomeSourceToSignUpControllerISpec.scala [hunk 3]
```diff
@@ -55,6 +56,28 @@ class WhatIncomeSourceToSignUpControllerISpec extends ComponentSpecBase  {
       }
     }
 
+    "redirect to task list" when {
```
Comment (aug24) [inline it/controllers/agent/WhatIncomeSourceToSignUpControllerISpec.scala:59]: New test.

File: it/controllers/individual/incomesource/WhatIncomeSourceToSignUpControllerISpec.scala [hunk 4]
```diff
@@ -61,6 +63,29 @@ class WhatIncomeSourceToSignUpControllerISpec extends ComponentSpecBase  {
       }
     }
 
+    "redirect to task list" when {
```
Comment (aug24) [inline it/controllers/individual/incomesource/WhatIncomeSourceToSignUpControllerISpec.scala:66]: New test.

File: scripts/start [hunk 5]
```diff
@@ -12,21 +12,12 @@ which nc &>/dev/null && nc -vz localhost $MONGO_PORT 2>/dev/null || die "Can't f
...
-   sm --start INCOME_TAX_SUBSCRIPTION "${@:2}"
-   sm --start INCOME_TAX_SUBSCRIPTION_SELF_EMPLOYED_FRONTEND "${@:2}"
-   sm --start INCOME_TAX_SUBSCRIPTION_STUBS "${@:2}"
-   sm --start INCOME_TAX_SUBSCRIPTION_ELIGIBILITY "${@:2}"
-   sm --start INCOME_TAX_SUBSCRIPTION_ELIGIBILITY_FRONTEND "${@:2}"
-   sm --start TAX_ENROLMENTS "${@:2}"
-else
-   sm --start ITSA_SUBSC_ALL "$@"
-fi
+sm --start ITSA_SUBSC_MIN "$@"
```
Comment (aug24) [inline scripts/start:15]: Just tidying up to use the new profile.

File: test/controllers/agent/WhatIncomeSourceToSignUpControllerSpec.scala [hunk 6]
```diff
@@ -48,7 +48,7 @@ class WhatIncomeSourceToSignUpControllerSpec extends AgentControllerBaseSpec
   override val authorisedRoutes: Map[String, Action[AnyContent]] = Map()
 
   "show" should {
-    "return 200 OK status" when {
+    "return 200 OK status if _anything_ is available and foreign property is allowed" when {
```
Comment (aug24) [inline test/controllers/agent/WhatIncomeSourceToSignUpControllerSpec.scala:51]: Clarify that this behaviour is specific to the foreign policy switch being on.

Comment (sb-dev) [reply test/controllers/agent/WhatIncomeSourceToSignUpControllerSpec.scala:51]: `_anything_` new syntax? 🧐

**Review by sb-dev — APPROVED** [review]

**Pull request merged**